youtube_dl/extractor/yledl.py

```diff
@@ -0,0 +1,75 @@
+# coding: utf-8
+from __future__ import unicode_literals
+from collections import defaultdict
+import subprocess
+import json
+
+from .common import InfoExtractor
+
+
+class YleDLIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:areena|arenan).yle.fi/(?P<id>[0-9]-[0-9]+)'
+    _GEO_COUNTRIES = ['FI']
+
+    _TEST = {
+        'url': 'https://areena.yle.fi/1-4256816',
+        'md5': 'b9658c5960a8c2ca4ba8f1b0ff079df2',
+        'info_dict': {
+            'id': '1_iq074q8b',
+            'ext': 'mxf',
+            'title': 'Luottomies | Luottomies jouluspeciaali',
+            'description':
+                'Tommia harmittaa kun sukulaiset ovat tulossa pilaamaan '
+                'mukavan perhejoulun. Muuttuuko mieli isosta yllätyksestä? '
+                'Joulun erikoisjakson on ohjannut Jalmari Helander.',
+            'upload_date': '20171207',
+            'height': 1080,
+            'width': 1920,
+            'fps': 25,
+            'duration': 1302,
+            'timestamp': 1512633989,
+            'extractor': 'Kaltura',
+            'uploader_id': 'ovp@yle.fi',
+            'webpage_url_basename': '1-4256816',
+            'webpage_url': 'https://areena.yle.fi/1-4256816'
+        }
+    }
+
+    def _real_extract(self, url):
+        props = {}
+
+        # Get essential data
+        props['id'] = self._match_id(url)
+
+        cp = subprocess.run(["yle-dl", "-q", "--showmetadata", url], capture_output=True, encoding='utf-8')
+        cp.check_returncode()
+        yledl = json.loads(cp.stdout)[0]
+
+        props['title'] = yledl.get('title', '<none>')
+        description = yledl.get('description', None)
+        if description:
+            props['description'] = description
+
+        ysubs = yledl.get('subtitles', [])
+        subtitles = defaultdict(list)
+        for s in ysubs:
+            subtitles[s['language']].append({'url': s['url'], 'ext': s['category']})
```

Comment:
You should be able to replace those 2 lines with
```python3
        subtitles[s['language']].extend({'url': s['url'], 'ext': s['category']} for s in ysubs)
```
. Untested but it should become a generator which gets implicitly iterated, which should be faster than appending 1 by 1.

Reply:
I like list comprehensions, but up to a limit. Your suggestion does not work as there is a reference to 's' outside the scope of the generator expression.

The code in this PR is outdated, due to the recent DMCA shenanigans. See the comment above for the current version.

Reply:
Doh, somehow I didn't realize the ref to `s` didn't exist if switching to that. My bad. Will need to use PyCharm 1st ...